lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1067,13 +1082,15 @@ namespace Js
             int i = 0;
             for (collation = uenum_next(collations, &collationLen, &status); collation != nullptr; collation = uenum_next(collations, &collationLen, &status))
             {
+                ICU_ASSERT(status, collation != nullptr && collationLen > 0);
                 if (strcmp(collation, "standard") == 0 || strcmp(collation, "search") == 0)
                 {
                     // continue does not create holes in ret because i is set outside the loop
                     continue;
                 }
 
-                const char *unicodeCollation = uloc_toUnicodeLocaleType("collation", collation);
+                // OS#17172584: OOM during uloc_toUnicodeLocaleType can make this return nullptr even for known collations
```

Comment:
I can't find anything for this bug number (?).

Reply:
https://microsoft.visualstudio.com/OS/_workitems/edit/17172584 ?

Reply:
Looks like VSO search is failing to find it for some reason...
Thanks for the link.

Reply:
From testing I can confirm that if ICU runs out of memory then you will indeed get back nullptr for known values.
So this is essentially yet another manifestation of the ICU data file OOM issue ( http://bugs.icu-project.org/trac/ticket/13712 ). 
However, what is rather unfortunate for this one is since the `uloc_toUnicodeLocaleType` API does not have an `UErrorCode` parameter/argument there is no way for you (the caller) to know that the failure was due to OOM, other than checking for getting back a nullptr.